modules/generic/src/main/scala/pureconfig/generic/EnumerationConfigReaderBuilder.scala

```diff
@@ -13,15 +15,19 @@ import shapeless.labelled._
  * @tparam A the type of objects capable of being read as an enumeration
  */
 trait EnumerationConfigReaderBuilder[A] {
-  def build(transformName: String => String): ConfigReader[A]
+  def build(transformName: String => String, validOptions: Set[String]): ConfigReader[A]
 }
 
 object EnumerationConfigReaderBuilder {
   implicit val deriveEnumerationReaderBuilderCNil: EnumerationConfigReaderBuilder[CNil] =
     new EnumerationConfigReaderBuilder[CNil] {
-      def build(transformName: String => String): ConfigReader[CNil] =
+      def build(transformName: String => String, validOptions: Set[String]): ConfigReader[CNil] =
         new ConfigReader[CNil] {
-          def from(cur: ConfigCursor): Result[CNil] = cur.failed(NoValidCoproductOptionFound(cur.value, Seq.empty))
+          def from(cur: ConfigCursor): Result[CNil] = {
+            val failure = ConvertFailure(KeyNotFound(cur.value.render(ConfigRenderOptions.concise()), validOptions), None, cur.path)
```

Comment:
This doesn't sound right... Enum configs are not about (object) keys, as they do not rely on objects at all. The true error is that we couldn't convert the value to an instance of the enum class, which is more accurately expressed as `CannotConvert(value in cursor, enum trait, some reason)`.

I wonder if `NoValidCoproductOptionFound` should be used here at all. It was built for coproduct options that have their own (possibly custom) readers, but here it seems overkill as we'd always generate something like:

```
- No valid coproduct option found for 'blue'.
  Can't use coproduct option 'blue':
    - Key not found: 'blue'. You might have a misconfigured ProductHint, since...
```

'blue' is here incorrectly used as a coproduct option and as an object key. This could be more concisely described as:

```
- Cannot convert 'blue' to Color: accepted values are 'rainy-blue' and 'sunny-yellow'.
```

What do you think, @leifwickland, @jcazevedo?

Reply:
As of https://github.com/pureconfig/pureconfig/pull/787/commits/b4bb7cf311cbf832d00b9c1dff51d16f9e1c952c, the error message printed when `Color` appears inside of a `Fooz` type looks like:
```
Cannot convert configuration to a pureconfig.EnumerationsSuite$Fooz. Failures are:
at 'foreground-color':
  - (e.conf: 2) Cannot convert "blue" to a Color. The valid options are:
      - rainy-blue
      - sunny-yellow
```

How do you feel about that approach?

Reply:
Looks good! 🙂